cdk_chalice/chalice.py

```diff
@@ -4,15 +4,41 @@
 import subprocess
 import sys
 import uuid
-from typing import Dict, Any
-
 import docker
+from dataclasses import dataclass
+from typing import Dict
 from aws_cdk import (
     aws_s3_assets as assets,
     core as cdk
 )
 
 
+@dataclass
+class DockerConfig:
+    """ Class for keeping all docker build configuration in one place,
+        use it in case your functions depend on packages that have
+        natively compiled dependencies, use this class to build the Chalice app
+        inside an AWS Lambda-like Docker container"""
+
+    # :param image: provide your docker image name, in case of empty will use default docker image
+    image: str
+
+    # :param env: environment variables to pass for docker container that build Chalice
+    env: dict
```

Comment:
Since there is an explicitly implemented __init__ method, I would ask to move the documentation of variable there, since Read the Docs won't be able to parse and display these directives well (this is where the documentation is deployed to on release). Type should be present in the docstring as well in order for Read the Docs to display it. Something like below (including the type hinting that both keys and values should be strings and not other objects - I omitted space after "str," on purpose, it is not parsed correctly otherwise):

from typing import Dict

"""
:param str image: Docker image name. 
    Defaults to image that closely mimics AWS Lambda execution environment.
:param Dict[str,str] env: Environment variables to set inside the container.
    AWS_DEFAULT_REGION is added and set to 'us-east-1' unless explicitly specified.
"""